mailpoet/lib/AutomaticEmails/WooCommerce/Events/AbandonedCart.php

```diff
@@ -165,11 +172,13 @@ private function scheduleAbandonedCartEmail(array $cartProductIds = []) {
       return;
     }
 
+    $this->wp->doAction(self::HOOK_SCHEDULE, $subscriber, $cartProductIds);
```

Comment:
The updates to this class confused me for a while. It says `scheduleAbandonedCartEmail`, so I'm like `Do we need an old abandoned cart email set up to make this work?`. But it turns out this rather "maybe" schedules an email, if there is one.

I think it may deserve a small refactoring — we could move all the subscriber fetch & fire hook logic somewhere to `lib/WooCommerce` and then use those hooks here and in automations.

Reply:
Will open a follow up ticket.